prompts/turbo.py

```diff
@@ -0,0 +1,95 @@
+import yaml
+from typing import Optional, List, Dict, Union
+from .prompt_builder import DynamicPrompt
+
+
+class TurboPrompt:
+    def __init__(
+        self,
+        system_prompt: Optional[Union[str, DynamicPrompt]] = None,
+        user_prompt: Optional[Union[str, DynamicPrompt]] = None,
+        assistant_prompt: Optional[Union[str, DynamicPrompt]] = None,
+        settings: Optional[Dict] = None,
+        title: Optional[str] = None,
+    ):
+        if system_prompt is None:
+            system_prompt = DynamicPrompt("<message>")
+        if user_prompt is None:
+            user_prompt = DynamicPrompt("<message>")
+        if assistant_prompt is None:
+            assistant_prompt = DynamicPrompt("<message>")
+        self.system_prompt = system_prompt
+        self.user_prompt = user_prompt
+        self.assistant_prompt = assistant_prompt
+        self.settings = settings
+        self.title = title
+        
+        self.prompts = []
+    
+    def add_user_message(self, **kwargs):
+        self._add_prompt("user", self.user_prompt.build(**kwargs))
+
+    def add_system_message(self, **kwargs):
+        self._add_prompt("system", self.system_prompt.build(**kwargs))
+
+    def add_assistant_message(self, **kwargs):
+        self._add_prompt("assistant", self.assistant_prompt.build(**kwargs))
+
+    def build(self) -> Dict[str, str]:
+        return [{prompt["type"]: prompt["prompt"]} for prompt in self.prompts]
+
+    def _add_prompt(self, prompt_type: str, prompt: Optional[Union[str, DynamicPrompt]] = None):
```

Comment:
Just remove the optional if you are going to raise an exception if it is not provided

Reply:
Okay. Ask chat wingman about this. :sweat_smile: